src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java

```diff
@@ -89,10 +89,15 @@ public FinancialTransactionDTO updateFinancialTransaction(
             @Valid FinancialTransactionUpdateDTO uDTO) {
 
         FinancialTransaction entity = financialTransactionRepository.findById(id)
-                .orElseThrow(() -> {
-                    throw new AppRuntimeException(ErrorCode.FT001,
-                            String.format("Financial transaction with id: %d not found", id));
-                });
+                .orElseThrow(() -> new AppRuntimeException(ErrorCode.FT001,
+                        String.format("Financial transaction with id: %d not found", id)));
+
+        if (uDTO.categoryId() != null) {
```

Comment:
Consider a situation where entity.categoryId!=null and uDTO.categoryId==null. Will the categoryId be updated?